features/step_definitions/helper_services.rb

```diff
@@ -155,32 +155,36 @@
 
 Given /^I have LDAP service in my project$/ do
     ###
-    # The original idea is trying to put ldap server in the openshift to make this flexy.
-    # Since we run the scenario in jenkins slave which is not in sdn, then two choices come to me:
+    # Since we run the scenario in jenkins agent which is not in sdn, then two choices come to me:
     # 1, Create a route for the ldapserver pod, but blocked by this us https://trello.com/c/9TXvMeS2 is done.
-    # 2, Port forward the ldap server pod to the jenkins slave.
+    # 2, Port forward the ldap server pod to the jenkins agent.
     # So take the second one since this one can be implemented currently
     ###
     step %Q/I run the :run client command with:/, table(%{
       | name  |ldapserver                             |
       | image |openshift/openldap-2441-centos7:latest |
       })
     step %Q/the step should succeed/
-    step %Q/I wait until replicationController "ldapserver-1" is ready/
+    step %Q/a pod becomes ready with labels:/, table(%{
+      | run=ldapserver |
+      })
 
     cb.ldap_pod = BushSlicer::Pod.get_labeled(["run", "ldapserver"], user: user, project: project).first
     cb.ldap_pod_name = cb.ldap_pod.name
     cache_pods cb.ldap_pod
 
+    step 'I obtain test data file "authorization/init.ldif"'
+    step %Q/the step should succeed/
+ 
     # Init the test data in ldap server.
-    @result = cb.ldap_pod.exec("bash", "-c", "curl -Ss https://raw.githubusercontent.com/openshift/origin/master/images/openldap/contrib/init.ldif | ldapadd -x -h 127.0.0.1 -p 389 -D cn=Manager,dc=example,dc=com -w admin", as: user)
+    @result = pod.exec("ldapadd", "-x", "-h", "127.0.0.1", "-p", "389", "-D", "cn=Manager,dc=example,dc=com", "-w", "admin", stdin: File.read(cb.test_file), as: user)
     step %Q/the step should succeed/
 
     # Port forword ldapserver to local
     step %Q/evaluation of `rand(32000...65536)` is stored in the :ldap_port clipboard/
     step %Q/I run the :port_forward background client command with:/, table(%{
-      | pod       | <%= cb.ldap_pod_name %> |
-      | port_spec | <%= cb.ldap_port %>:389 |
+      | pod       | <%= pod.name %>        |
+      | port_spec | <%= cb.ldap_port %>:389  |
```

Comment:
Please fix indentation here (and possibly other similar places). We use 2 spaces for indentation. In Cucumber the keyword `Examples: ` is indented the same as the `Scenario` line. Tables are indented deeper by 2 characters. Multi-line strings are indented at the same level  as the step they are par of (not deeper).

We now have also in the project a [EditorConfig](http://editorconfig.org/) file. If provides an editor agnostic way to set some styling settings like indentation type and width and removal of trailing white spaces. Please check the linked website to find whether your editor is supported.

For generic code style rules and more information, more style information, please check [coding_guide.adoc](https://github.com/openshift/cucushift/blob/v3/doc/coding_guide.adoc)

For vim specific settings you can see [doc/vim.adoc](https://github.com/openshift/cucushift/blob/v3/doc/vim.adoc).

-- saved reply

Reply:
looks nice except for the above

Reply:
I edited it in place so no worries.